src/raftstore/store/peer_storage.rs

```diff
@@ -919,26 +881,14 @@ pub fn do_snapshot(mgr: SnapManager, snap: &DbSnapshot, region_id: u64) -> raft:
 
     snapshot.mut_metadata().set_conf_state(conf_state);
 
-    let mut snap_file = try!(mgr.rl().get_snap_file(&key, true));
-    if snap_file.exists() {
-        if let Err(e) = snap_file.reader().and_then(|mut r| r.validate()) {
-            error!("[region {}] file {} is invalid, will regenerate: {:?}",
-                   region_id,
-                   snap_file.path().display(),
-                   e);
-            try!(snap_file.try_delete());
-            try!(snap_file.init());
-            try!(build_snap_file(&mut snap_file, snap, state.get_region()));
-        }
-    } else {
-        try!(build_snap_file(&mut snap_file, snap, state.get_region()));
-    }
+    let mut creator = try!(mgr.rl().get_send_snapshot_file_creator(&key));
```

Comment:
maybe `gen_snapshot_file_creator(&key, SnapFileType::SENDING)` is more clear than `get_send_snapshot_file_creator`